pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/GuardLogStatementRule.java

```diff
@@ -237,7 +240,14 @@ private boolean isUnguardedAccessOk(ASTMethodCall call, int messageArgIndex) {
         // so that we can ignore it
         return call.getArguments().toStream()
                    .drop(messageArgIndex) // remove the level argument if needed
-                   .all(it -> it instanceof ASTStringLiteral || it instanceof ASTLambdaExpression || it instanceof ASTVariableAccess || it instanceof ASTMethodReference);
+                   .all(GuardLogStatementRule::isDirectAccess);
+    }
+
+    private static boolean isDirectAccess(ASTExpression it) {
+        return it instanceof ASTStringLiteral || it instanceof ASTLambdaExpression
+                || it instanceof ASTVariableAccess || it instanceof ASTMethodReference
+                || it instanceof ASTFieldAccess || it instanceof ASTThisExpression
+                || (it instanceof ASTArrayAccess && isDirectAccess(((ASTArrayAccess) it).getQualifier()));
```

Comment:
Checking that the qualifier is a direct access should also be done for method references and field accesses (all QualifiableExpressions) : `foo().field`, `foo()::ref`

Reply:
I think it's illegal to use anything else than a class type as the LHS of a this expression, so I'm not sure it's worth checking for that